components/list/list.js

```diff
@@ -320,7 +320,7 @@ class List extends PageableMixin(SelectionMixin(LitElement)) {
 	_handleListItemAddButtonClick(e) {
 		e.stopPropagation();
 		/** Dispatched when the add button directly after the item is clicked. Event detail includes the key of the item directly above where the add button was clicked. */
-		this.dispatchEvent(new CustomEvent('d2l-list-add-button-click', { detail: { key: e.target.key } }));
+		this.dispatchEvent(new CustomEvent('d2l-list-add-button-click', { detail: { key: e.target.key, isFirstItem: e.detail.isFirstItem } }));
```

Comment:
This might be a bit weird - we are using the same event and key from list for clicking `button-add` above or below the first item (and then only for below for subsequent items). I wasn't sure/couldn't find if there was precedent for a different way of doing this but let me know any suggestions.

Reply:
So the consumer will listen for the one event (which seems good). But for the first item, they need to know whether an item is being added before or after. I suppose we could have `e.detail.position: 'before|after'`, and we default to `after`. 🤷‍♂️

Reply:
I like the position idea - it makes the consumer have to be aware that position is used.